crates/orderbook/src/solvable_orders.rs

```diff
@@ -136,6 +141,8 @@ impl SolvableOrdersCache {
         let db_solvable_orders = self.database.solvable_orders(min_valid_to).await?;
         let orders = filter_banned_user_orders(db_solvable_orders.orders, &self.banned_users);
         let orders = filter_unsupported_tokens(orders, self.bad_token_detector.as_ref()).await?;
+        let orders =
+            filter_invalid_signature_orders(orders, self.signature_validator.as_ref()).await;
```

Comment:
The code below here skips checking balances if the block hasn't changed since the previous update. The same is true for the validity of signatures so as an optimization we could skip that too. However it doesn't fit in as nicely so not sure if we should do it.
I'm also wondering if validating signatures when we have a lot of these kind of orders could be a problem if it is done every block. We could say we cache this for a short time but again not sure if worth it.

Reply:
This is a good point. I think the easier way to deal with this is to have a `CachedSignatureValidator` that only invalidates its cache on new blocks (this way we can keep complexity out of the `update` fn).

Reply:
Created #312